tests/test_duet.py

```diff
@@ -0,0 +1,228 @@
+        benchmark_peak_indices = np.load(peak_indices_path)
+
+        duet_peak_indices = nussl.utils.find_peak_indices(benchmark_hist, duet.num_sources,
+                                                          threshold=duet.peak_threshold,
+                                                          min_dist=[duet.attenuation_min_distance,
+                                                                    duet.delay_min_distance])
+
+        assert np.all(benchmark_peak_indices == duet_peak_indices)
+
+    def test_convert_peaks(self):
+        duet = nussl.Duet(self.signal, 3)
+        duet.stft_ch0 = self.benchmark_dict['benchmark_stft_ch0']
+        duet.stft_ch1 = self.benchmark_dict['benchmark_stft_ch1']
+        duet.frequency_matrix = self.benchmark_dict['benchmark_wmat']
+        duet.symmetric_atn = self.benchmark_dict['benchmark_sym_atn']
+        duet.delay = self.benchmark_dict['benchmark_delay']
+        duet.attenuation_bins = self.benchmark_dict['benchmark_atn_bins']
+        duet.delay_bins = self.benchmark_dict['benchmark_delay_bins']
+        duet.peak_indices = self.benchmark_dict['benchmark_peak_indices']
+
+        delay_peak_path = os.path.join('duet_reference', 'duet_benchmarks', 'benchmark_delay_peak.npy')
+        atn_peak_path = os.path.join('duet_reference', 'duet_benchmarks', 'benchmark_atn_peak.npy')
+        atn_delay_est_path = os.path.join('duet_reference', 'duet_benchmarks', 'benchmark_atn_delay_est.npy')
+        benchmark_delay_peak = np.load(delay_peak_path)
+        benchmark_atn_peak = np.load(atn_peak_path)
+        benchmark_atn_delay_est = np.load(atn_delay_est_path)
+
+        delay_peak, atn_delay_est, atn_peak = duet._convert_peaks()
+
+        assert np.all(benchmark_delay_peak == delay_peak)
+        assert np.all(benchmark_atn_delay_est == atn_delay_est)
+        assert np.all(benchmark_atn_peak == atn_peak)
+
+    def test_compute_masks(self):
+        duet = nussl.Duet(self.signal, 3)
+        duet.stft_ch0 = self.benchmark_dict['benchmark_stft_ch0']
+        duet.stft_ch1 = self.benchmark_dict['benchmark_stft_ch1']
+        duet.frequency_matrix = self.benchmark_dict['benchmark_wmat']
+        duet.symmetric_atn = self.benchmark_dict['benchmark_sym_atn']
+        duet.delay = self.benchmark_dict['benchmark_delay']
+        duet.attenuation_bins = self.benchmark_dict['benchmark_atn_bins']
+        duet.delay_bins = self.benchmark_dict['benchmark_delay_bins']
+        duet.peak_indices = self.benchmark_dict['benchmark_peak_indices']
+        duet.delay_peak = self.benchmark_dict['benchmark_delay_peak']
+        duet.atn_peak = self.benchmark_dict['benchmark_atn_peak']
+
+        mask_path = os.path.join('duet_reference', 'duet_benchmarks', 'benchmark_masks.npy')
+        benchmark_mask = np.load(mask_path)
+
+        masks = duet._compute_masks()
+        for i in range(len(masks)):
+            assert np.all(benchmark_mask[i].mask == masks[i].mask)
+
```

Comment:
One more test case I want you to write: Instantiate a `Duet` object, run it, then run the same object again on a different audio signal. Make sure it doesn't crash and the audio sounds fine in both. (I've had some issues with running the same object on two different). Do it like this:
 `duet = nussl.Duet(signal1)`
`duet.run()`
`duet.audio_signal = signal2`
`duet.run()`
Etc, but fill in the blanks wrt setting up the input and output signals.